salt/grains/core.py

```diff
@@ -2554,7 +2554,11 @@ def ip6_interfaces():
                 iface_ips.append(inet["address"])
         for secondary in ifaces[face].get("secondary", []):
             if "address" in secondary:
-                iface_ips.append(secondary["address"])
+                try:
+                    socket.inet_pton(socket.AF_INET6, secondary["address"])
+                    iface_ips.append(secondary["address"])
+                except OSError:
```

Comment:
Note that python2 version of `socket.inet_pton()` raises `socket.error` instead of `OSError` here.

Reply:
The "master" branch is already not Python 2 compatible. OTOH I have pushed the fix for this to avoid breaking core grains just because of this

Reply:
Actually, `pre-commit` checks are now failing because of adding `socket.error`. I'm reverting it.

Reply:
Salt no longer supports Python 2.